pkg/logs/internal/tailers/file/rotate_nix.go

```diff
@@ -37,8 +38,19 @@ func (t *Tailer) DidRotate() (bool, error) {
 		return true, nil
 	}
 
+	lastReadOffset := t.lastReadOffset.Load()
+	fileOneSize := fi1.Size()
+
 	recreated := !os.SameFile(fi1, fi2)
-	truncated := fi1.Size() < t.lastReadOffset.Load()
+	truncated := fileOneSize < lastReadOffset
+
+	if recreated {
+		log.Debugf("File rotation detected due to recreation, f1: %v, f2: %v", fi1, fi2)
+	}
+
+	if truncated {
```

Comment:
```suggestion
	else if truncated {
```
Otherwise most of the time both messages will be logged: new rotated files are usually smaller than the ones they replace.